cooperator/models/company.py

```diff
@@ -138,3 +143,59 @@ def onchange_financial_risk_approval_required(self):
     def onchange_generic_rules_approval_required(self):
         if self.generic_rules_approval_required:
             self.display_generic_rules_approval = True
+
+    def _accounting_data_initialized(self):
+        return self.chart_template_id or self.env[
+            "account.chart.template"
+        ].existing_accounting(self)
+
+    def _init_cooperator_data(self):
+        """
+        Generate default cooperator data for the company.
+        """
+        # this method exists to initialize data correctly for the following
+        # possible cases:
+        # 1. when a new company is created. in this case it will be called
+        #    when the account chart template is loaded.
+        # 2. when a database is initialized with only the cooperator module,
+        #    and the l10n_generic_coa module is loaded after it by the
+        #    post_init_hook of the account module. in this case it is first
+        #    called by the post_init_hook of this module (cooperator) (but
+        #    with no effect, as explained below) and then again when the
+        #    account chart template is loaded.
+        # 3. when the cooperator module is installed on an existing database.
+        #    in this case it is called by the post_init_hook of this module.
+        subscription_request_model = self.env["subscription.request"]
+        for company in self:
+            if not company._accounting_data_initialized():
+                # if no account chart template has been loaded yet and no
+                # accounting data exists, then no accounting data should be
+                # created yet because all existing journals and accounts will
+                # be deleted when the account chart template will be loaded.
+                # this method will be called again after the account chart
+                # template has been loaded.
+                continue
+            subscription_request_model.create_journal(company)
+        # this is called here to support the first 2 cases explained above.
+        self._init_cooperator_demo_data()
```

Comment:
I'm confused by the `demo` word here. Why `demo`? The first 2 cases can happen in production, right?
Did you mean something like `default`?

Reply:
i understand your confusion. indeed, it is about actual demo data, and this will be called in production, but `_init_cooperator_demo_data()` has an early return in case demo data is disabled, and it has no effect in that case. do you see another way to support the case where a database is initialized with `cooperator` and demo data but no l10n module? or do you have a suggestion for a comment explaining the situation better maybe?

Reply:
there is no demo journal (it is the regular one), but indeed, the demo cooperator account (which must be created manually without demo data) gets erased when loading the account chart template. this is why it is created from here.